### ISSUE TITLE
Error in `perl': double free or corruption

### ISSUE THREAD

**Issue by m-jackson** [lstein/Perl-GD#8]
GD Warning: gd-png error: no colors in palette
**\* Error in `perl': double free or corruption (!prev): 0x00000000026851f0 ***
Aborted (core dumped)

**Comment by wd** [comment]
I got the same issue,  and code snippet like this, see the attachment to find a.png.
```
my $img = GD::Image->new('a.png');
my $width = $img->width();
my $height = $img->height();

my $dest = new GD::Image(90,90);
$dest->copyResampled($img, 0, 0, 0, 0, 90, 90, $width, $height);

print $dest->png; # crash here
```

![a](https://cloud.githubusercontent.com/assets/58177/14349706/7a0998d0-fcf7-11e5-82e4-b54c0f6b99b8.png)

**Comment by rurban** [comment]
I see. We don't do any error handling at all yet. The problem is by returned empty data.
I'll stabilize it

**Comment by rurban** [comment]
Fixed with 2e978bdcb676b434836620c986487a9ed6ad89a1
for 2.59
